httptools/parser.pyi

```diff
@@ -1,80 +1,88 @@
-from typing import TYPE_CHECKING
+from typing import Optional
 
-if TYPE_CHECKING:
-	from httptools.parser.parser import URL
 
 class HttpRequestParser:
-    def __init__(self, protocol):
-        """HttpRequestParser
-
-        protocol -- a Python object with the following methods
-        (all optional):
-
-          - on_message_begin()
-          - on_url(url: bytes)
-          - on_header(name: bytes, value: bytes)
-          - on_headers_complete()
-          - on_body(body: bytes)
-          - on_message_complete()
-          - on_chunk_header()
-          - on_chunk_complete()
-          - on_status(status: bytes)
-        """
-        pass
-
-    def get_http_version(self) -> str:
-        """Return an HTTP protocol version."""
-        pass
-
-    def should_keep_alive(self) -> bool:
-        """Return ``True`` if keep-alive mode is preferred."""
-        pass
-
-    def should_upgrade(self) -> bool:
-        """Return ``True`` if the parsed request is a valid Upgrade request.
+	def __init__(self, protocol):
+		"""HttpRequestParser
```

Comment:
I think you accidentally converted this file to use tabs. Please use spaces (as per pep8)